### PR TITLE
[Docs] Add pydocstringformatter to precommits

### LINKED ISSUE #165: Consider adding pydocstringformatter to pre-commits
Adding [pydocstringformatter](https://pypi.org/project/pydocstringformatter/#:~:text=A%20tool%20to%20automatically%20format%20Python%20docstrings%20to%20follow%20recommendations,for%20currently%20supported%20auto%2Dformatting) to pre-commits would help building consistent docs.

### PR DESCRIPTION
**Pull request by Roland-djee**
Closes #165.

### CODE REVIEW

**Comment by Roland-djee** [comment]
@dominikandreasseitz Not sure the gain is over the Moon but still.

**Review by nmheim — APPROVED** [review]
Looks good to me:)

File: qadence/extensions.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def _validate_diff_mode(backend: Backend, diff_mode: DiffMode) -> None:
 
 
 def _set_backend_config(backend: Backend, diff_mode: DiffMode) -> None:
-    """_summary_
+    """_summary_.
```
Comment (nmheim) [inline qadence/extensions.py:62]: Hehe:)

File: pyproject.toml [hunk 2]
```diff
@@ -78,6 +78,7 @@ dependencies = [
   "black",
   "isort",
   "ruff",
+  "pydocstringformatter",
```
Comment (dominikandreasseitz) [inline pyproject.toml:81]: can we get a test executable to locally run the docstringformatter? like test-examples, test-docs etc

Comment (Roland-djee) [reply pyproject.toml:81]: `pre-commit run --all-files` ?

**Review by dominikandreasseitz — APPROVED** [review]
lgtm

File: qadence/extensions.py (hunk 1 above)
Comment (Roland-djee) [reply qadence/extensions.py:62]: Yeah I know. It is a bit dumb overall. Not entirely convinced it's going to bring much but hey.

**Pull request merged**